src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -200,40 +204,45 @@ public function get(): ?array
                 $sql .= ' '.$this->driverConnection->getDatabasePlatform()->getWriteLockSQL();
             }
 
-            $doctrineEnvelope = $this->executeQuery(
+            $doctrineEnvelopeList = $this->executeQuery(
                 $sql,
                 $query->getParameters(),
                 $query->getParameterTypes()
-            )->fetchAssociative();
+            )->fetchAllAssociative();
 
-            if (false === $doctrineEnvelope) {
+            if ([] === $doctrineEnvelopeList) {
                 $this->driverConnection->commit();
                 $this->queueEmptiedAt = microtime(true) * 1000;
 
-                return null;
+                return [];
             }
             // Postgres can "group" notifications having the same channel and payload
             // We need to be sure to empty the queue before blocking again
             $this->queueEmptiedAt = null;
 
-            $doctrineEnvelope = $this->decodeEnvelopeHeaders($doctrineEnvelope);
+            $this->lastDeliveredButNotYetHandledEnvelopesIds = $this->extractDoctrineEnvelopeListIds($doctrineEnvelopeList);
 
             $queryBuilder = $this->driverConnection->createQueryBuilder()
                 ->update($this->configuration['table_name'])
                 ->set('delivered_at', '?')
-                ->where('id = ?');
-            $now = new \DateTimeImmutable('UTC');
-            $this->executeStatement($queryBuilder->getSQL(), [
-                $now,
-                $doctrineEnvelope['id'],
-            ], [
-                Types::DATETIME_IMMUTABLE,
-            ]);
+                ->where($this->driverConnection->createQueryBuilder()->expr()->in('id', '?'));
+            $this->executeStatement(
+                $queryBuilder->getSQL(),
+                [
+                    new \DateTimeImmutable('UTC'),
+                    $this->lastDeliveredButNotYetHandledEnvelopesIds,
+                ],
+                [
+                    Types::DATETIME_IMMUTABLE,
+                    class_exists(ArrayParameterType::class) ? ArrayParameterType::INTEGER : Types::SIMPLE_ARRAY,
```

Comment:
This is wrong. `Types::SIMPLE_ARRAY` is something totally different. The fallback for older DBAL versions is `Connection::PARAM_INT_ARRAY`.

And the code below does not check for older versions, so it is inconsistent.